protocol/arp.go

```diff
@@ -91,5 +96,10 @@ func (a *ARP) UnmarshalBinary(data []byte) error {
 	n += int(a.HWLength)
 	a.IPDst = make([]byte, a.ProtoLength)
 	copy(a.IPDst, data[n:n+int(a.ProtoLength)])
+	n += int(a.ProtoLength)
+	if len(data[n:]) > 0 {
```

Comment:
ditto. No need to copy the "padding" bytes, since it is not a valid case that a component consumes the padding bytes.